### ISSUE TITLE
Marking an inline-by-defaults element to be a block

### ISSUE THREAD

**Issue by coderlex** [pugjs/pug#2525]
I like Jade and wish to use it as native markup language before building my HTML templates. Of course they should pretty look on output without a mess with displaying elements. The only problem I faced was about inline-by-defaults elements like in following case:

```html
<div class="header">
	<a class="header__logo" href="#">
		<img src="img/demo/header-logo.png" alt="SiteName">
	</a>
	<div class="header__something"></div>
</div>
```

In the example the first `a` tag considered to be inline and is rendered in a line right after `.header` opening tag, but sematically it's a block and must be rendered as above.

So is there a way to specify how an element must behave in individual cases (i.e. inline / block)?

**Comment by TimothyGu** [comment]
No, there isn't. It would complicate the syntax, and for performance reasons we don't construct a full HTML DOM that would allow us to semantically deduce the block/inline distinction of an element.

FWIW, we plan to deprecate pretty mode. If precise prettifying is important to you, you could consider using a proper beautifier such as [jsbeautify](https://www.npmjs.com/package/js-beautify).
